--- a/ssh/tcpip.go	Thu Oct 17 08:54:15 2013 +1100
+++ b/ssh/tcpip.go	Fri Oct 25 21:19:05 2013 +0200
@@ -299,7 +299,7 @@
 	if err := c.writePacket(marshal(msgChannelOpen, channelOpenDirectMsg{
 		ChanType:      "direct-tcpip",
 		PeersId:       ch.localId,
-		PeersWindow:   1 << 14,
+		PeersWindow:   1 << 24,
 		MaxPacketSize: 1 << 15, // RFC 4253 6.1
 		raddr:         raddr,
 		rport:         uint32(rport),
